docs/pages/versions/unversioned/sdk/webbrowser.md

```diff
@@ -60,30 +65,82 @@ Opens the url with Safari in a modal on iOS using `SFAuthenticationSession`. The
+
+### `WebBrowser.warmUpAsync(browserPackage)`
+
+_Android only_
+
+This method calls `warmUp` method on [CustomTabsClient](<https://developer.android.com/reference/android/support/customtabs/CustomTabsClient.html#warmup(long)>) for specified package.
+
+#### Arguments
+
+- **browserPackage (_string_)** -- **optional** -- package of browser to be warmed up. If not set, preferred browser will be warmed.
+
+#### Returns
+
+A promise resolving with `{ type: 'warming', package: string }`
+
+### `WebBrowser.mayInitWithUrlAsync(url, package)`
+
+_Andrdoi only_
+
+This method initiates (if needed) [CustomTabsSession](https://developer.android.com/reference/android/support/customtabs/CustomTabsSession.html#maylaunchurl) and calls its `mayLaunchUrl` method for browser specified by the package.
+
+#### Arguments
+
+- **url (_string_)** -- url of page that user is likely to be loaded firts
+- **package (_string_)** -- **optional** -- package of browser to be informed. If not set, preferred browser will be used.
+
+#### Returns
+
+The promise resolves with `{ type: 'mayInitWithUrl', package: string, url: string }`.
+
+### `WebBrowser.coolDownAsync(browserPackage)`
+
+_Andrdoi only_
+
+This methods removes all bindings to services created by [`warmUpAsync`](#webbrowserwarmupasyncnbrowserpackage) or [`mayInitWithUrlAsync`](#webbrowseramayinitwithurlsyncurl-package). You should call this method once you don't need them to avoid potential memory leaks. However, those binding would be cleared once your application is destroyed, which might be sufficient in most cases.
+
+#### Arguments
+
+- **browserPackage (_string_)** -- **optional** -- package of browser to be cooled. If not set, preferred browser will be used.
+
+#### Returns
+
+The promise resolves with `{ type: 'cooling', package: string }` when cooling is performed, or `{ result: 'Nothing to cool down' }` when there was no connection to be dismissed.
 
 ### `WebBrowser.dismissBrowser()`
 
-Dismisses the system's presented web browser. On Android calling this method doesn't actually do anything, the user still has to manually dismiss the browser.
+_iOS only_
+
+Dismisses the presented web browser.
 
 #### Returns
 
 The promise resolves with `{ type: 'dismiss' }`.
 
 ### `WebBrowser.getCustomTabsSupportingBrowsers`
 
-Available only on Android.
+_Android only_
 
-Returns two lists of applications package names supporting Custom Tabs. Under `packages` key you can find a list of all the applications. Under `default` key only the default applications will be listed. Note that list might not be fully reliable, as it uses `PackageManager.getResolvingActivities`.
+Returns a list of applications package names supporting Custom Tabs, Custom Tabs service, user chosen and prefered one. This may not be fully reliable, since it uses `PackageManager.getResolvingActivities` under the hood. (For example, some browsers might not be present in `browserPackages` list once another browser is set to defult.)
```

Comment:
```suggestion
Returns a list of applications package names supporting Custom Tabs, Custom Tabs service, user chosen and preferred one. This may not be fully reliable, since it uses `PackageManager.getResolvingActivities` under the hood. (For example, some browsers might not be present in `browserPackages` list once another browser is set to default.)
```